packages/daemon/src/daemon.js

```diff
@@ -665,28 +731,81 @@ const makeEndoBootstrap = (
       petStoreFormulaIdentifier,
     );
 
+    const identifyOnFormula = (formula, propertyName, namePath) => {
+      if (
+        allowedInspectorFormulaIdentificationsByType[formula.type] === undefined
+      ) {
+        // Cannot provide a reference under the requested formula type.
+        return { formulaIdentifier: undefined, remainderPath: namePath };
+      }
+      const allowedInspectorFormulaIdentifications =
+        allowedInspectorFormulaIdentificationsByType[formula.type];
+      if (!allowedInspectorFormulaIdentifications.includes(propertyName)) {
+        // Cannot provide a reference for the requested property.
+        return { formulaIdentifier: undefined, remainderPath: namePath };
+      }
+      if (formula.type === 'eval' && propertyName === 'endowments') {
+        // We consume an additional part of the path for the endowment name.
+        const [endowmentName, ...namePathRest] = namePath;
+        assertPetName(endowmentName);
+        const endowmentIndex = formula.names.indexOf(endowmentName);
+        if (endowmentIndex === -1) {
+          // Cannot provide a reference to the missing endowment.
+          return { formulaIdentifier: undefined, remainderPath: namePath };
+        }
+        const formulaIdentifier = formula.values[endowmentIndex];
+        return { formulaIdentifier, remainderPath: namePathRest };
+      }
+      const formulaIdentifier = formula[propertyName];
+      return { formulaIdentifier, remainderPath: namePath };
+    };
+
+    /** @type {import('./types.js').IdentifyFn} */
+    const identify = async maybeNamePath => {
+      const namePath = Array.isArray(maybeNamePath)
+        ? maybeNamePath
+        : [maybeNamePath];
+      const [petName, propertyName, ...namePathRest] = namePath;
+      assertPetName(petName);
+      assertPetName(propertyName);
+      const entryFormulaIdentifier = petStore.identifyLocal(petName);
+      if (entryFormulaIdentifier === undefined) {
+        return undefined;
+      }
+      const { type: formulaType, number: formulaNumber } =
+        parseFormulaIdentifier(entryFormulaIdentifier);
```

Comment:
By the time we get here, we may have removed the type from the formula identifier, but it can be trivially recovered from the controller object for the number.

Reply:
When we have a formula for every formula identifier (just the number), it might make sense to just generally store that on the controller too, so we benefit from the memo and avoid redundant reads from the formula store.